packages/sdk/src/core/types/wallet.types.ts

```diff
@@ -0,0 +1,49 @@
+export interface WalletResponse<D> {
+  code: number;
+  status: WalletResponseStatus;
+  type: WalletResponseType;
+  message: string;
+  data: D | null;
+}
+
+export enum WalletResponseStatus {
+  SUCCESS = 'success',
+  FAILURE = 'failure',
+}
+
+export type WalletResponseType = WalletResponseSuccessType | WalletResponseFailedType | WalletResponseRejectedType;
+
+export enum WalletResponseSuccessType {
+  CONNECTION_SUCCESS = 'CONNECTION_SUCCESS',
+  GET_ACCOUNT_SUCCESS = 'GET_ACCOUNT_SUCCESS',
+  SIGN_SUCCESS = 'SIGN_SUCCESS',
+  ADD_NETWORK_SUCCESS = 'ADD_NETWORK_SUCCESS',
+  SWITCH_NETWORK_SUCCESS = 'SWITCH_NETWORK_SUCCESS',
+  TRANSACTION_SUCCESS = 'TRANSACTION_SUCCESS',
+}
+
+export enum WalletResponseFailedType {
+  NOT_CONNECTED = 'NOT_CONNECTED',
+  UNRESOLVED_TRANSACTION_EXISTS = 'UNRESOLVED_TRANSACTION_EXISTS',
+  INVALID_FORMAT = 'INVALID_FORMAT',
+  WALLET_LOCKED = 'WALLET_LOCKED',
+  ACCOUNT_MISMATCH = 'ACCOUNT_MISMATCH',
+  NO_ACCOUNT = 'NO_ACCOUNT',
+  TRANSACTION_FAILED = 'TRANSACTION_FAILED',
+  SIGN_FAILED = 'SIGN_FAILED',
+  ALREADY_CONNECTED = 'ALREADY_CONNECTED',
+  NETWORK_TIMEOUT = 'NETWORK_TIMEOUT',
+  REDUNDANT_CHANGE_REQUEST = 'REDUNDANT_CHANGE_REQUEST',
+  NETWORK_ALREADY_EXISTS = 'NETWORK_ALREADY_EXISTS',
+  UNADDED_NETWORK = 'UNADDED_NETWORK',
+  UNSUPPORTED_TYPE = 'UNSUPPORTED_TYPE',
+  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
+}
```

Comment:
should we unify those types in the future in a common package used by the main Adena repo and this SDK ?

Reply:
Yes, I would like to unify the types and responses.
If the SDK and wallet responses were separate, it wouldn't be a problem if the provider did the mapping for me, but I'd have to manage them separately and always consider what I'm missing.